runtime/master/src/main/java/org/apache/nemo/runtime/master/scheduler/LambdaScheduler.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.nemo.runtime.master.PlanStateManager;
+import org.apache.nemo.runtime.master.resource.ExecutorRepresenter;
+import org.apache.reef.annotations.audience.DriverSide;
+import org.apache.reef.tang.Injector;
+import org.apache.reef.tang.JavaConfigurationBuilder;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.wake.remote.ports.TcpPortProvider;
+import org.apache.reef.wake.remote.ports.parameters.TcpPortRangeBegin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.NotThreadSafe;
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A simple scheduler for batch job with Nemo Lambda Executor.
+ * - Keeps track of new executors
+ * - Schedules all tasks in the plan at once.
+ * - Crashes the system upon any other events (should be fixed in the future)
+ * - Never stops running.
+ */
+@DriverSide
+@NotThreadSafe
+public final class LambdaScheduler implements Scheduler {
+  private static final Logger LOG = LoggerFactory.getLogger(StreamingScheduler.class.getName());
+
+  private final PlanStateManager planStateManager;
+
+  @Inject
+  private LambdaScheduler(final TaskDispatcher taskDispatcher,
+                     final PendingTaskCollectionPointer pendingTaskCollectionPointer,
+                     final ExecutorRegistry executorRegistry,
+                     final PlanStateManager planStateManager,
+                     final PipeManagerMaster pipeManagerMaster) {
+    this.planStateManager = planStateManager;
+    LOG.info("Using Lambda Scheduler");
+  }
+
+  @Override
+  public void schedulePlan(final PhysicalPlan submittedPhysicalPlan,
+                           final int maxScheduleAttempt) {
+
+    if (maxScheduleAttempt > 1) {
+      throw new UnsupportedOperationException();
+    }
+
+    planStateManager.updatePlan(submittedPhysicalPlan, maxScheduleAttempt);
+    planStateManager.storeJSON("submitted");
+
+    // We presume serverless function has been deployed and uploaded
+
+    // Prepare tasks
+    final List<Stage> allStages = submittedPhysicalPlan.getStageDAG().getTopologicalSort();
+    allStages.stream().forEach(stageToSchedule -> {
+      LOG.info("Execute new stage");
+      final List<StageEdge> stageIncomingEdges =
+        submittedPhysicalPlan.getStageDAG().getIncomingEdgesOf(stageToSchedule.getId());
+      final List<StageEdge> stageOutgoingEdges =
+        submittedPhysicalPlan.getStageDAG().getOutgoingEdgesOf(stageToSchedule.getId());
+
+      // the list of maps between vertex ID and readables.
+      final List<Map<String, Readable>> vertexIdToReadables = stageToSchedule.getVertexIdToReadables();
+      final List<String> taskIdsToSchedule = planStateManager.getTaskAttemptsToSchedule(stageToSchedule.getId());
+
+
+      taskIdsToSchedule.forEach(taskId -> {
+        AWSLambda client = AWSLambdaClientBuilder.standard().build();
```

Comment:
We don't have to create a client per each task. Why don't you reuse it?

Reply:
Indeed. This is a great idea. 
I'll reuse the AWS client.